lib/ansible/playbook/play_context.py

```diff
@@ -359,9 +359,10 @@ def set_task_and_variable_override(self, task, variables, templar):
             for variable_name in variable_names:
                 if attr in attrs_considered:
                     continue
-                if isinstance(delegated_vars, dict) and variable_name in delegated_vars:
-                    setattr(new_info, attr, delegated_vars[variable_name])
-                    attrs_considered.append(attr)
+                if delegated_vars:
```

Comment:
Took another look at the logic setting delegated_vars -- I think this check needs to be
```
 if task.delegate_to is not None
```
Otherwise if the delegated_to host has no vars, I think it would still end up with an empty dict on Line 318 and therefore this condition won't be triggered.